internal/runbits/requirements/requirements.go

```diff
@@ -209,6 +222,61 @@ func (r *RequirementOperation) ExecuteRequirementOperation(
 		pg = nil
 	}
 
+	if r.Auth.Authenticated() && operation == bpModel.OperationAdded && ns.Type() == model.NamespacePackage {
+		pg = output.StartSpinner(out, locale.Tr("progress_cve_search", requirementName), constants.TerminalAnimationInterval)
+
+		vulnerabilities, err := model.FetchVulnerabilitiesForIngredient(r.Auth, &request.Ingredient{
+			Namespace: ns.String(),
+			Name:      requirementName,
+			Version:   requirementVersion,
+		})
+		if err != nil {
+			return errs.Wrap(err, "Failed to retrieve vulnerabilities")
+		}
+
+		var safe bool
+		if vulnerabilities == nil || (vulnerabilities != nil && vulnerabilities.Vulnerabilities.Length() == 0) {
+			safe = true
+		}
+
+		if !safe {
+			pg.Stop(locale.T("progress_unsafe"))
+			pg = nil
+
+			if r.shouldPromptForSecurity(vulnerabilities) {
+				out.Print("")
+				cont, err := r.promptForSecurity(out, vulnerabilities)
+				if err != nil {
+					return errs.Wrap(err, "Failed to prompt for security")
+				}
+
+				if !cont {
+					return locale.NewError("err_pkgop_security_prompt", "Operation aborted due to security prompt")
+				}
+			} else {
+				var severityBreakdown []string
+				if len(vulnerabilities.Vulnerabilities.Critical) > 0 {
+					severityBreakdown = append(severityBreakdown, fmt.Sprintf("[RED]%d Critical[/RESET]", len(vulnerabilities.Vulnerabilities.Critical)))
+				}
+				if len(vulnerabilities.Vulnerabilities.High) > 0 {
+					severityBreakdown = append(severityBreakdown, fmt.Sprintf("[ORANGE]%d high[/RESET]", len(vulnerabilities.Vulnerabilities.High)))
+				}
+				if len(vulnerabilities.Vulnerabilities.Medium) > 0 {
+					severityBreakdown = append(severityBreakdown, fmt.Sprintf("[YELLOW]%d medium[/RESET]", len(vulnerabilities.Vulnerabilities.Medium)))
+				}
+				if len(vulnerabilities.Vulnerabilities.Low) > 0 {
+					severityBreakdown = append(severityBreakdown, fmt.Sprintf("[MAGENTA]%d low[/RESET]", len(vulnerabilities.Vulnerabilities.Low)))
+				}
+
+				out.Print("    " + strings.TrimSpace(locale.Tr("warning_vulnerable", strconv.Itoa(vulnerabilities.Vulnerabilities.Length()), strings.Join(severityBreakdown, ", "))))
+			}
+		} else {
+			pg.Stop(locale.T("progress_safe"))
+		}
+
+		pg = nil
```

Comment:
Did you consider splitting this into its own function instead of continuing to add to this already-ginormous function? :)

Reply:
I did not, only because this is following the pattern that already exists in the function. I know we've talked about refactoring this function before and there should be a story for it somewhere. I think that's a more appropriate place to start breaking this flow up.